components/url/lib.rs

```diff
@@ -163,6 +165,12 @@ impl ServoUrl {
 
 impl fmt::Display for ServoUrl {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        if self.0.to_string().len() > 40 {
```

Comment:
Ok, I tried it myself and this will work (https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7a33776e9070188d2d4073b5c05b6683):

`if self.0.as_str().len() > 40`